grammars/css.cson

```diff
@@ -20,294 +20,1218 @@
+        'patterns': [
+          {
+            'include': '#comment-block'
+          }
+          {
+            'include': '#escapes'
+          }
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @media
+        'begin': '(?i)(?=@media(?:\\s|\\(|$|/\\*))'
+        'end': '\\s*(})|(?=;)'
+        'endCaptures':
+          '1':
+            'name': 'punctuation.section.property-list.end.css'
+        'patterns': [
+          {
+            'begin': '(?i)(@)media(?=\\s|\\(|$|/\\*)'
+            'beginCaptures':
+              '0':
+                'name': 'keyword.control.at-rule.media.css'
+              '1':
+                'name': 'punctuation.definition.keyword.css'
+            'end': '\\s*(?=[;{])'
+            'name': 'meta.at-rule.media.css'
+            'patterns': [
+              {
+                'include': '#media-query-list'
+              }
+            ]
+          }
+          {
+            'begin': '\\s*({)'
+            'beginCaptures':
+              '1':
+                'name': 'punctuation.section.property-list.begin.css'
+            'end': '(?=})'
+            'patterns': [
+              {
+                'include': '$self'
+              }
+            ]
+          }
+        ]
+      }
+      {
+        # @page
+        'begin': '(?i)(@)page(?=[\\s:{;]|$|/\\*)'
+        'captures':
+          '0':
+            'name': 'keyword.control.at-rule.page.css'
+          '1':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '\\s*(?=$|[{:;])'
+        'name': 'meta.at-rule.page.css'
+        'patterns': [
+          {
+            'include': '#rule-list'
+          }
+        ]
+      }
+      {
+        # @counter-style
+        'begin': '(?i)((@)counter-style)(?=[\\s{\'";]|/\\*|$)\\s*'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.counter-style.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'contentName': 'variable.parameter.style-name.css'
+        'end': '(?={|[@;])'
+        'name': 'meta.at-rule.counter-style.css'
+        'patterns': [
+          {
+            'include': '#comment-block'
+          }
+          {
+            'include': '#escapes'
+          }
+        ]
+      }
+      {
+        # @document
+        'begin': '(?i)((@)document)(?:\\s+|$|(?=[\'";{]|/\\*))'
+        'beginCaptures':
+          '1':
+            'name': 'keyword.control.at-rule.document.css'
+          '2':
+            'name': 'punctuation.definition.keyword.css'
+        'end': '{|(?=[@;])'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.section.document-styles.begin.css'
+        'name': 'meta.at-rule.document.css'
+        'patterns': [
+          {
+            'begin': '(?i)\\s*(?<![\\w-])(url-prefix|domain|regexp)(\\()\\s*'
+            'beginCaptures':
+              '1':
+                'name': 'support.function.${1:/downcase}-rule.css'
```

Comment:
Will change to `support.function.document-rule.css` instead. I just wanted some way to identify the functions as specific to `@document` blocks.